### PR TITLE
Feature: message eth sign + Address Book Migration

### LINKED ISSUE #961: Idex doesn't work properly
image1

image2

App Information:
App Name: MetaMask
App Version: 0.2.1 (233)
Installed App Version: 0.2.1 (233)

Device Information:
Device: iPhone8,1
iOS Version: 12.3.1
Language: de-DE (German (Germany))
Carrier: Telekom.de
Timezone: MESZ
Architecture: arm64
Connection Status: Cellular data
Paired Apple Watch: N/A

### LINKED ISSUE #1109: Add support for `eth_sign`
Previously reported in #169, but it seems like only `personal_sign` was implemented.

Should be coupled with the big scary warning like we use in desktop:
https://github.com/MetaMask/metamask-extension/blob/e0e06709d3ea79db3830bf1e68667bc41b1fa732/app/_locales/en/messages.json#L1559

### PR DESCRIPTION
**Pull request by estebanmino**
**Description**

Add support for eth_sign.

BLOCKED BY GABA

**Checklist**

* [ ] There is a related GitHub issue
* [ ] Tests are included if applicable
* [ ] Any added code is fully documented

**Issue**

Resolves #961 
Resolves #1109

### CODE REVIEW

**Comment by ibrahimtaveras00** [comment]
**Issue 1:**

on Android device, tapping on learn more next to the security message, nothing happens, seen here = 

seen here = http://recordit.co/LtyaVQUfsT

it doesn’t open the zendesk article like in iOS

**Issue 2:**

I wanted to test wallet connect signing, but when I'm on Android, on the wallet view and tap on the scanner icon at top right, app crashes 

seen here = http://recordit.co/bwX2XQuf9B

**Issue 3:** 

On iOS, I just sent eth from this account to ibrahimtester.eth and when I tap the drop down I don't see it appear in the list

Screen Shot 2019-10-17 at 8 23 53 PM

Screen Shot 2019-10-17 at 8 24 03 PM

On Android, app crashes immediately when attempting to send 

Seen here = https://recordit.co/Thz9Iixz0R

**Comment by ibrahimtaveras00** [comment]
Fixes look good on both OS's, QA passed 👍

**Review by brunobar79 — APPROVED** [review]
This looks good. One minor comment, and @ibrahimtaveras00  make sure you run the app on develop first, then rebuild (without reinstalling) to test the address book.  Previous contacts should still be available.

**Comment by ibrahimtaveras00** [comment]
Last test case looks good 👍 

Sent some tokens to address ending in E69a and ibrahimtester.eth on develop, then after building this branch, double checked and addresses were still displayed in the dropdown, = http://recordit.co/28rnZffWSC

File: app/store/index.js [hunk 1]
```diff
@@ -1,13 +1,31 @@
...
+	}
+};
+
 const persistConfig = {
 	key: 'root',
+	version: 0,
 	storage: AsyncStorage,
-	stateReconciler: autoMergeLevel2 // see "Merge Process" section for details.
+	stateReconciler: autoMergeLevel2, // see "Merge Process" section for details.
+	migrate: createMigrate(migrations, { debug: true })
```
[the remark this answers is not in the document]
Comment (estebanmino) [reply app/store/index.js]: good catch

**Pull request merged**